services/services.go

```diff
@@ -644,34 +635,36 @@ func getIndexPageData() (*types.IndexPageData, error) {
 	}
 	data.ScheduledCount = scheduledCount
 
+	// get epochs from db
 	var epochs []*types.IndexPageDataEpochs
-	err = db.ReaderDb.Select(&epochs, `SELECT epoch, finalized , eligibleether, globalparticipationrate, votedether FROM epochs ORDER BY epochs DESC LIMIT 15`)
+	err = db.ReaderDb.Select(&epochs, `SELECT epoch, finalized, globalparticipationrate, votedether FROM epochs ORDER BY epochs DESC LIMIT 10`)
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving index epoch data: %v", err)
 	}
-	epochsMap := make(map[uint64]bool)
+	minEpoch := epochs[len(epochs)-1].Epoch
+	epochsMap := make(map[uint64]*types.IndexPageDataEpochs)
+	// set epoch struct values and add to map
 	for _, epoch := range epochs {
+		if data.EpochParticipationRate == 0 && epoch.GlobalParticipationRate < 1 {
+			data.EpochParticipationRate = epoch.GlobalParticipationRate
+		}
 		epoch.Ts = utils.EpochToTime(epoch.Epoch)
-		epoch.FinalizedFormatted = utils.FormatYesNo(epoch.Finalized)
-		epoch.VotedEtherFormatted = utils.FormatBalance(epoch.VotedEther, currency)
-		epoch.EligibleEtherFormatted = utils.FormatEligibleBalance(epoch.EligibleEther, currency)
-		epoch.GlobalParticipationRateFormatted = utils.FormatGlobalParticipationRate(epoch.VotedEther, epoch.GlobalParticipationRate, currency)
-		epochsMap[epoch.Epoch] = true
+		epoch.FinalizedFormatted = utils.FormatEpochStatus(epoch.Finalized, epoch.GlobalParticipationRate)
+		//epoch.VotedEtherFormatted = utils.FormatBalance(epoch.VotedEther, currency)
+		//epoch.EligibleEtherFormatted = utils.FormatEligibleBalance(epoch.EligibleEther, currency)
+		epoch.GlobalParticipationRateFormatted = utils.FormatGlobalParticipationRateStyle(epoch.VotedEther, epoch.GlobalParticipationRate, currency)
+		epoch.ExecutionReward = big.NewInt(0)
+		epochsMap[epoch.Epoch] = epoch
 	}
 
-	var blocks []*types.IndexPageDataBlocks
-	err = db.ReaderDb.Select(&blocks, `
+	// get slots from db
+	var slots []*types.IndexPageDataBlocks
+	err = db.ReaderDb.Select(&slots, `
```

Comment:
This SQL statement should be limited to 10 rows as well, should it not?

Reply:
As you noted, it may be possible for multiple slots to have the same slot number. Although I haven't seen this happen in practice, the old code did check for that possibility. To avoid potential issues, I didn't want to limit the slotsMap to only 10 rows.